tests/headers/basicauth_test.go

```diff
@@ -180,3 +163,70 @@ func TestBasicAuth(t *testing.T) {
 		})
 	}
 }
+
+func TestBasicAuthOrdering(t *testing.T) {
+	// It is not clear from the documentation of http.Request.BasicAuth()
```

Comment:
Please rephrase a bit e.g:

The documentation of `http.Request.BasicAuth()` doesn't clearly specify whether only the first Authorization header is used and the other ones are ignored. This could potentially lead to security issues if two HTTP servers that look at different headers are chained together.

The desired behavior would be to decline(?) an HTTP request with more than one Authorization header.

or something along those lines

Reply:
Thanks, that is more clear! :)
But I don't agree on this phrasing:
`The desired behavior would be to decline(?) an HTTP request with more than one Authorization header.`
We don't want to decline the entire request just because there are more than one Authorization header. We just want the `BasicAuth` function to give an error of some kind (or for the golang team to improve their documentation). Since `BasicAuth` already returns an `ok` bool status I thought we could just set this to `false`.
Or do you think it should be handled differently?

Reply:
I think that returning a `Bad Request` status would be acceptable in case of double auth header.
Note that we will likely ban basic auth so the behavior of BasicAuth here is not relevant and [the spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.8) doesn't seem to have an opinion on the matter.

Reply:
Ok, then I will change the desired behavior to be a `Bad Request`